functions/habapp.bash

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC2181
+
+## Function for installing HABApp (python3 rule engine)
+## Valid arguments: "install" or "remove"
+##
+##    habapp_setup()
+##
+habapp_setup() {
+  local InstallFolder=/opt
+  local VenvName=habapp
+  local ConfigFolder=/etc/openhab/habapp
+
+  if [[ $1 == "remove" ]]; then
+    echo -n "$(timestamp) [openHABian] Removing HABApp service ... "
+    if ! cond_redirect systemctl stop habapp.service; then echo "FAILED (stop service)"; return 1; fi
+    if ! rm -f /etc/systemd/system/habapp.service; then echo "FAILED (remove service)"; return 1; fi
+    if cond_redirect systemctl -q daemon-reload &> /dev/null; then echo "OK"; else  echo "FAILED (daemon-reload)"; return 1; fi
+
+    echo -n "$(timestamp) [openHABian] Uninstalling HABApp ... "
+    if rm -rf "${InstallFolder:?}/${VenvName:?}/"; then echo "OK"; else echo "FAILED"; return 1; fi
+
+    if [[ -n "$INTERACTIVE" ]]; then
+      whiptail --title "HABApp removed" --msgbox "We permanently removed the HABApp installation from your system." 7 80
+    fi
+    return 0;
+  fi
+  if [[ $1 != "install" ]]; then return 1; fi
+
+  echo -n "$(timestamp) [openHABian] Installing HABApp prerequisites (pip, venv) ... "
+  if cond_redirect apt-get install --yes python3-pip python3-dev python3-venv; then echo "OK"; else echo "FAILED"; return 1; fi
+
+  echo -n "$(timestamp) [openHABian] Creating venv ... "
+  if ! cond_redirect cd "$InstallFolder"; then echo "FAILED"; return 1; fi
+  if cond_redirect python3 -m venv "$VenvName"; then echo "OK"; else echo "FAILED"; return 1; fi
+  
+  echo -n "$(timestamp) [openHABian] Installing HABApp ... "
+  if ! cond_redirect cd "$VenvName"; then echo "FAILED"; return 1; fi
+  if ! cond_redirect source bin/activate; then echo "FAILED"; return 1; fi
+
+  if ! cond_redirect python3 -m pip install --upgrade pip habapp; then echo "FAILED"; return 1; fi
+  if cond_redirect deactivate; then echo "OK"; else echo "FAILED"; return 1; fi
+
+  echo -n "$(timestamp) [openHABian] Creating HABApp configuration folder ... "
+  if ! cond_redirect mkdir -p "$ConfigFolder"; then echo "FAILED"; return 1; fi
+  if ! cond_redirect chown -R openhab:openhab /etc/openhab/habapp; then echo "FAILED"; return 1; fi
```

Comment:
group openhabian preferred

Reply:
Is it openhabian:openhabian or openhabian:openhab?
Both have write access to the openhab-conf folder, right?

Reply:
openhab:openhabian
The standard "openhabian" or other admin Linux user isn't necessarily in the openhab group so may not have group access rights.
The openhab user has access anyway by user rights.